server/com.dexels.navajo.compiler.tsl/src/com/dexels/navajo/compiler/tsl/internal/BundleQueueComponent.java

```diff
@@ -170,18 +173,39 @@ private void checkForChangedScripts(Event e, String folder) {
                     	continue;
                     }
                     enqueueScript(scriptName, changedScript);
-                    enqueueDependentScripts(scriptName, new HashSet<String>());
                 }
             } catch (IllegalArgumentException e1) {
                 logger.warn("Error in handling changed script {}: {}", changedScript, e1);
             }
         }
     }
 
+    // check whether dependencies of the current script are satisfied, and if not create them on demand
+    private void checkOnDemandScriptDependencies(String script) {
+        String rpcName = script;
+        String bareScript = script.substring(script.lastIndexOf('/') + 1);
+        if (bareScript.indexOf('_') >= 0) {
+            rpcName = script.substring(0, script.lastIndexOf('_'));
+        }
 
-    private void enqueueDependentScripts(String script, Set<String> history) {
-        history.add(script);
-        
+        // For now, only entity dependencies are relevant script dependencies.
+        // For instance, in the case the server runs in DEVELOP_MODE where
+        // entities are lazily loaded, all bundles for super entities also need
+        // to be installed
+        List<Dependency> dependencies = depanalyzer.getDependencies(rpcName, Dependency.ENTITY_DEPENDENCY);
```

Comment:
I would suggest to have the dependency type state whether it is needed here or not so that when we need to add another dependency type for this situation it is done at the dependency definition and not here